app/controllers/locations_controller.rb

```diff
@@ -1,25 +1,29 @@
 class LocationsController < ApplicationController
   def create
-    @location = build_location
+    @profile = current_user
+
+    if save_location?
+      enqueue_geocode_job
 
-    if update_location?
       redirect_to profile_url
     else
-      @profile = current_user
-
       render "profiles/show"
     end
   end
 
   private
 
-  def update_location?
-    @location.update(location_params) &&
-      current_user.update(zipcode: location_params[:zipcode])
+  def enqueue_geocode_job
+    GeocodeJob.perform_later(@location)
```

Comment:
Did you want to access this directly as an instance variable? Should it use the `build_location` helper so you know `@location` has been initialised?